tests/integration/issues/github_241/test_topk.py

```diff
@@ -0,0 +1,59 @@
+import os
+
+import numpy as np
+import pytest
+from jina.flow import Flow
+from jina.proto import jina_pb2
+from jina.drivers.helper import array2pb
+
+
+@pytest.fixture
+def config(tmpdir):
+    os.environ['JINA_TOPK_DIR'] = str(tmpdir)
+    os.environ['JINA_NDOCS'] = '3'
+    os.environ['JINA_TOPK'] = '9'
+    os.environ['JINA_TOPK_OVERRIDE'] = '11'
+
+
+def random_docs(num_docs, embed_dim=10, jitter=1):
+    for j in range(num_docs):
+        d = jina_pb2.Document()
+        d.tags['id'] = j
+        d.text = b'hello'
+        d.embedding.CopyFrom(array2pb(np.random.random([embed_dim + np.random.randint(0, jitter)])))
+        yield d
+
+
+def validate_results(resp):
+    assert len(resp.search.docs) == int(os.environ['JINA_NDOCS'])
+    for doc in resp.search.docs:
+        assert len(doc.matches) == int(os.environ['JINA_TOPK'])
+
+
+def test_topk(config):
```

Comment:
Can you do the same test again with `shards = 2`. I would guess, this test will fail then. I think, the implementation in the `gateway.py` should use a `SliceQL` instead of a `VectorSearchDriver`.

Reply:
Assuming, this is kind of a `hack`, `VectorSearchDriver` is almost always there in a `Flow` while `SliceQL` not so common (and `SliceQL` does not read `top_k` parameter). 

This is just a way to make it easier from the ouside, but is not really the best way to do things. (It is kind of a syntax sugar for a common pattern)